common/canonical_serialization/README.md

```diff
@@ -0,0 +1,471 @@
+# Libra Canonical Serialization (LCS)
+
+## Overview
+
+This document defines Libra Canonical Serialization (LCS). LCS defines a deterministic means for translating a message or data structure into bytes irrespective of platform, architecture, or programming language.
+
+## Background
+
+In Libra, participants pass around messages or data structures that often times need to be signed by one party and verified by many. Serialization is the process of converting a message into a byte array. Most serialization approaches support loose standards such that two implementations can produce two different byte streams that would represent the identical message. As a result, messages that are passed between participants must either retain the original serialized bytes or risk losing the ability to verify the message. This creates a burden programmatically by requiring the code to maintain a copy of the bytes after deserializing the object often leads to confusion about the safety and correctness of this approach. While there exist a handful of existing deterministic serialization formats, e.g. XDR, there exists no mainstream, well maintained, and available deterministic serialization. To address this, we propose Libra Canonical Serialization that defines a deterministic means for translating a message into bytes.
```

Comment:
I wouldn't say "Most serialization approaches..." but that is certainly true for "Many serialization approaches"

"This creates a burden programmatically by requiring the code to maintain a copy of the bytes after deserializing the object often leads to confusion about the safety and correctness of this approach."

Suggest deleting programmatically and inserting "and this" before often.

Reply:
Ah, yes, good point, this can be cleaned up to be more readable!